packages/oid4vci-holder/src/agent/OID4VCIHolderService.ts

```diff
@@ -193,21 +196,28 @@ export const verifyCredentialToAccept = async (args: VerifyCredentialToAcceptArg
 }
 
 export const verifyCredential = async (args: IVerifyCredentialArgs, context: RequiredContext): Promise<VerificationResult> => {
-  // We also allow/add boolean, because 4.x Veramo returns a boolean for JWTs. 5.X will return better results
-  const result: IVerifyResult | boolean = (await context.agent.verifyCredential(args)) as IVerifyResult | boolean
+  const { credential } = args
+
+  return (typeof credential === 'string' && CredentialMapper.isSdJwtEncoded(credential))
+    ? await verifySDJWTCredential({ credential }, context)
+    : await verifyW3CCredential(args, context)
+}
+
+export const verifyW3CCredential = async (args: IVerifyCredentialArgs, context: RequiredContext): Promise<CredentialVerificationResult> => {
+  const { policies } = args
+
+  const result: IVerifyResult | boolean = (await context.agent.verifyCredential(args))
 
   if (typeof result === 'boolean') {
     return {
-      source: CredentialMapper.toWrappedVerifiableCredential(args.credential as OriginalVerifiableCredential),
       result,
       ...(!result && {
         error: 'Invalid JWT VC',
-        errorDetails: `JWT VC could was not valid with policies: ${JSON.stringify(args.policies)}`,
-      }),
-      subResults: [],
+        errorDetails: `JWT VC was not valid with policies: ${JSON.stringify(policies)}`,
+      })
     }
   } else {
-    const subResults: Array<VerificationSubResult> = []
+    //TODO look at what this is doing and make it simple and readable
```

Comment:
Please leave existing things in place. Sub results can come from for instance status list verification happening

Reply:
ok but they are not using nor filled ever

Reply:
Not in this library no. But as mentioned trustlist implementations can fill these objects. This ensures we expose the interfaces as how they can be used by other implementations

Reply:
reverted